tests/bdp_spark.erl

```diff
@@ -0,0 +1,138 @@
+    ok = remove_spark_service(Node1, ?SERVICE_2, []),
+
+    pass.
+
+
+create_spark_bucket_types(Node) ->
+    rt:create_and_activate_bucket_type(Node, <<"strong">>, [{consistent, true}]),
+    rt:create_and_activate_bucket_type(Node, <<"maps">>, [{datatype, map}]),
+    lager:info("Spark bucket types 'strong' and 'maps' created"),
+    ok.
+
+
+add_spark_service(Node, ServiceName, Config, WaitServices) ->
+    ok = bdp_util:service_added(Node, ServiceName, ?SPARK_MASTER_TYPE, Config),
+    ok = bdp_util:wait_services(Node, {[], WaitServices}),
+    lager:info("Service definition ~p (~s) added to node ~p", [ServiceName, ?SPARK_MASTER_TYPE, Node]),
+    ok.
+
+
+start_services(Node, [{ServiceNode, ServiceName} | Rest]) ->
+    ok = bdp_util:service_started(Node, ServiceNode, ServiceName, ?SPARK_MASTER_TYPE),
+    lager:info("Service ~p up on ~p node", [ServiceName, ServiceNode]),
+    start_services(Node, Rest);
+start_services(_, []) -> ok.
+
+
+stop_services(Node, [{ServiceNode, ServiceName} | Rest]) ->
+    ok = bdp_util:service_stopped(Node, ServiceNode, ServiceName, ?SPARK_MASTER_TYPE),
+    lager:info("Service ~p down on ~p", [ServiceName, Node]),
+    stop_services(Node, Rest);
+stop_services(_, []) -> ok.
+
+
+remove_spark_service(Node, ServiceName, WaitServices) ->
+    ok = bdp_util:service_removed(Node, ServiceName),
+    ok = bdp_util:wait_services(Node, {[], WaitServices}),
+    lager:info("Service ~p removed from ~p", [ServiceName, Node]),
+    ok.
+
+
+test_spark_fail_recovery() ->
+    %% execute $ ps -ef | grep org.apache.spark.deploy.master.Master
+    %% to find out the PIDs of running masters and to make sure they are actually running
+    %%
+    %% To find out which of the masters is the leader open spark
+    %% master log files located in the spark-master/logs if the line
+    %%
+    %% ‘INFO RiakEnsembleLeaderElectionAgent: We have lost leadership’
+    %%
+    %% is found as a last one from RiakEnsembleLeaderElectionAgent,
+    %% this instance is currently a standby master. Let’s ‘tail -f’
+    %% this log.  This way we’ll see when it’ll gain leadership
+
+    %% Go to the instance that has and entry
+    %% ‘INFO RiakEnsembleLeaderElectionAgent: We have gained leadership’
+    %% and execute ‘kill <master’s pid>’
+
+    %% Verify that in the standby master’s logs that you tail there are entries like
+    %% 15/07/09 02:13:46 INFO RiakEnsembleLeaderElectionAgent: We have gained leadership
+    %% 15/07/09 02:13:46 INFO Master: I have been elected leader! New state: RECOVERING
+
+    [DevPath | _] = rtdev:devpaths(),
+
+    Node1Path = DevPath ++ "/dev/dev" ++ integer_to_list(1),
+    Node2Path = DevPath ++ "/dev/dev" ++ integer_to_list(2),
+
+    lager:info("Node1 path = ~s, Node2 path = ~s", [Node1Path, Node2Path]),
+
+    LogSubPath = "/lib/data_platform-1/priv/spark-master/logs/spark-" ++ ?SPARK_IDENT_STRING ++ "-org.apache.spark.deploy.master.Master-1-" ++ ?HOSTNAME ++  ".out",
```

Comment:
When a devrel is built, there's no guarantee that the data_platform directory will be "data_platform-1". If stagedevrel is uses, the directory is just "data_platform". Once we starting using real versions, it will be "data_platform-$vsn"; further indicating that this would need to be updated for each time data_platform's version changes. Using "data_platform*" should work.